hollow-test/src/test/java/com/netflix/hollow/test/consumer/TestHollowConsumerTest.java

```diff
@@ -98,4 +108,112 @@ public void testAddSnapshot_afterUpdate() throws Exception {
         consumer.triggerRefresh();
         assertEquals(version2, consumer.getCurrentVersionId());
     }
+
+    @Test
+    public void testDelta_versionTransition() throws Exception {
+        long snapshotVersion = 1L;
+        TestHollowConsumer consumer = new TestHollowConsumer.Builder()
+                .withBlobRetriever(new TestBlobRetriever())
+                .build();
+
+        HollowWriteStateEngine snapshotWriteStateEngine = new HollowWriteStateEngineBuilder().build();
+        consumer.addSnapshot(snapshotVersion, snapshotWriteStateEngine);
+        assertEquals("Should be no version", VERSION_NONE, consumer.getCurrentVersionId());
+        consumer.triggerRefreshTo(snapshotVersion);
+        assertEquals("Should be at snapshot version", snapshotVersion, consumer.getCurrentVersionId());
+
+        long deltaToVersion = 2L;
+        HollowWriteStateEngine deltaWriteStateEngine = new HollowWriteStateEngineBuilder().build();
```

Comment:
calling them snapshot and delta write state engine is confusing IMO, since they're really "complete" write state engines